yaksh/views.py

```diff
@@ -3191,3 +3191,106 @@ def download_course_progress(request, course_id):
     for student in stud_details:
         writer.writerow(student)
     return response
+
+
+@login_required
+@email_verified
+def course_forum(request, course_id):
+    user = request.user
+    base_template = 'user.html'
+    moderator = False
+    if is_moderator(user):
+        base_template = 'manage.html'
+        moderator = True
+    course = get_object_or_404(Course, id=course_id)
+    if (not course.is_creator(user) and not course.is_teacher(user)
+            and not course.is_student(user)):
+        raise Http404('You are not enrolled in {0} course'.format(course.name))
+    if 'search' in request.GET:
+        search_term = request.GET['search']
+        posts = course.post.filter(active=True, title__icontains=search_term)
```

Comment:
If search results in more than 10 threads then pagination will not work for the search results. Clicking on the page number redirects back to the original list.
Maybe create a separate view function for searching posts or add a search request parameter in the paginator.html file.
Also, add `get_queryset()` before filter method. I mean `course.post.get_queryset().filter(...)`